sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Literal, Optional, Sequence, Tuple
+
+import pytz
+from hazelcast.client import HazelcastClient
+from hazelcast.core import HazelcastJsonValue
+from pydantic import StrictStr
+
+from feast import Entity, FeatureView, RepoConfig
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel
+from feast.usage import log_exceptions_and_usage
+
+
+class HazelcastOnlineStoreConfig(FeastConfigBaseModel):
+    """Online store config for Hazelcast store"""
+
+    type: Literal["hazelcast"] = "hazelcast"
+    """Online store type selector"""
+
+    cluster_name: StrictStr = "dev"
+    """Name of the cluster you want to connect. The default cluster name is `dev`"""
+
+    cluster_members: Optional[List] = ["localhost:5701"]
+    """List of member addresses which is connected to your cluster"""
+
+    discovery_token: Optional[StrictStr] = ""
+    """The discovery token of your Hazelcast Viridian cluster"""
+
+    ssl_cafile_path: Optional[StrictStr] = ""
+    """Absolute path of CA certificates in PEM format."""
+
+    ssl_certfile_path: Optional[StrictStr] = ""
+    """Absolute path of the client certificate in PEM format."""
+
+    ssl_keyfile_path: Optional[StrictStr] = ""
+    """Absolute path of the private key file for the client certificate in the PEM format."""
+
+    ssl_password: Optional[StrictStr] = ""
+    """Password for decrypting the keyfile if it is encrypted."""
+
+    key_ttl_seconds: Optional[int] = 0
+    """Hazelcast key bin TTL (in seconds) for expiring entities"""
+
+
+class HazelcastOnlineStore(OnlineStore):
+    _client: Optional[HazelcastClient] = None
+
+    def _get_client(self, config: HazelcastOnlineStoreConfig):
+        logging.getLogger("hazelcast").setLevel(logging.ERROR)
+        if not self._client:
```

Comment:
It's better to use `is` with this check:
```
if self._client is None:
```

Reply:
I understand that but none of other OS implementations does not provide this, I thought they are ignoring this because it's a rare condition. 

Should we care and implement Singleton pattern here? Maybe I can ask to Feast community about why they don't request this in OS implementations.